### ISSUE TITLE
Duplicate resources - sentry-external-modules.txt

### ISSUE THREAD

**Issue by dawidhyzy** [getsentry/sentry-android-gradle-plugin#412]
### Gradle Version

7.5.1

### AGP Version

7.4.0-rc01

### Code Minifier/Optimizer

R8

### Version

3.3.0

### Sentry SDK Version

6.9.1

### Steps to Reproduce

1. Build a project

### Expected Result

Project builds successfully

### Actual Result

Project build fails with:
```
* Exception is:
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':myproject:mergeStagingDebugAssets'.
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.lambda$executeIfValid$1(ExecuteActionsTaskExecuter.java:142)
	at org.gradle.internal.Try$Failure.ifSuccessfulOrElse(Try.java:282)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeIfValid(ExecuteActionsTaskExecuter.java:140)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:128)
	at org.gradle.api.internal.tasks.execution.CleanupStaleOutputsExecuter.execute(CleanupStaleOutputsExecuter.java:77)
	at org.gradle.api.internal.tasks.execution.FinalizePropertiesTaskExecuter.execute(FinalizePropertiesTaskExecuter.java:46)
	at org.gradle.api.internal.tasks.execution.ResolveTaskExecutionModeExecuter.execute(ResolveTaskExecutionModeExecuter.java:51)
	at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:57)
	at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:56)
	at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:36)
	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.executeTask(EventFiringTaskExecuter.java:77)
	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:55)
	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:52)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:204)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:199)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:66)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:59)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:157)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:59)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.call(DefaultBuildOperationRunner.java:53)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:73)
	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter.execute(EventFiringTaskExecuter.java:52)
	at org.gradle.execution.plan.LocalTaskNodeExecutor.execute(LocalTaskNodeExecutor.java:69)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:327)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:314)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:307)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:293)
	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.execute(DefaultPlanExecutor.java:420)
	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.run(DefaultPlanExecutor.java:342)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
	at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48)
Caused by: com.android.build.gradle.tasks.ResourceException: /Users/dawidhyzy/Developer/myproject-android-tv/myproject/build/generated/assets/sentry/stagingDebug/sentry-external-modules.txt	/Users/dawidhyzy/Developer/myproject-android-tv/myproject/build/generated/assets/sentry/stagingDebug/sentry-external-modules.txt: Error: Duplicate resources
	at com.android.build.gradle.tasks.MergeSourceSetFolders.doFullTaskAction(MergeSourceSetFolders.kt:173)
	at com.android.build.gradle.tasks.MergeSourceSetFolders.doTaskAction(MergeSourceSetFolders.kt:122)
	at com.android.build.gradle.internal.tasks.NewIncrementalTask$taskAction$$inlined$recordTaskAction$1.invoke(BaseTask.kt:69)
	at com.android.build.gradle.internal.tasks.Blocks.recordSpan(Blocks.java:51)
	at com.android.build.gradle.internal.tasks.NewIncrementalTask.taskAction(NewIncrementalTask.kt:46)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:125)
	at org.gradle.api.internal.project.taskfactory.IncrementalInputsTaskAction.doExecute(IncrementalInputsTaskAction.java:32)
	at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:51)
	at org.gradle.api.internal.project.taskfactory.AbstractIncrementalTaskAction.execute(AbstractIncrementalTaskAction.java:25)
	at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:29)
	at org.gradle.api.internal.tasks.execution.TaskExecution$3.run(TaskExecution.java:236)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$1.execute(DefaultBuildOperationRunner.java:29)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$1.execute(DefaultBuildOperationRunner.java:26)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:66)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:59)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:157)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:59)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.run(DefaultBuildOperationRunner.java:47)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:68)
	at org.gradle.api.internal.tasks.execution.TaskExecution.executeAction(TaskExecution.java:221)
	at org.gradle.api.internal.tasks.execution.TaskExecution.executeActions(TaskExecution.java:204)
	at org.gradle.api.internal.tasks.execution.TaskExecution.executeWithPreviousOutputFiles(TaskExecution.java:187)
	at org.gradle.api.internal.tasks.execution.TaskExecution.execute(TaskExecution.java:165)
	at org.gradle.internal.execution.steps.ExecuteStep.executeInternal(ExecuteStep.java:89)
	at org.gradle.internal.execution.steps.ExecuteStep.access$000(ExecuteStep.java:40)
	at org.gradle.internal.execution.steps.ExecuteStep$1.call(ExecuteStep.java:53)
	at org.gradle.internal.execution.steps.ExecuteStep$1.call(ExecuteStep.java:50)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:204)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:199)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:66)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:59)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:157)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:59)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.call(DefaultBuildOperationRunner.java:53)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:73)
	at org.gradle.internal.execution.steps.ExecuteStep.execute(ExecuteStep.java:50)
	at org.gradle.internal.execution.steps.ExecuteStep.execute(ExecuteStep.java:40)
	at org.gradle.internal.execution.steps.RemovePreviousOutputsStep.execute(RemovePreviousOutputsStep.java:68)
	at org.gradle.internal.execution.steps.RemovePreviousOutputsStep.execute(RemovePreviousOutputsStep.java:38)
	at org.gradle.internal.execution.steps.CancelExecutionStep.execute(CancelExecutionStep.java:41)
	at org.gradle.internal.execution.steps.TimeoutStep.executeWithoutTimeout(TimeoutStep.java:74)
	at org.gradle.internal.execution.steps.TimeoutStep.execute(TimeoutStep.java:55)
	at org.gradle.internal.execution.steps.CreateOutputsStep.execute(CreateOutputsStep.java:51)
	at org.gradle.internal.execution.steps.CreateOutputsStep.execute(CreateOutputsStep.java:29)
	at org.gradle.internal.execution.steps.CaptureStateAfterExecutionStep.executeDelegateBroadcastingChanges(CaptureStateAfterExecutionStep.java:124)
	at org.gradle.internal.execution.steps.CaptureStateAfterExecutionStep.execute(CaptureStateAfterExecutionStep.java:80)
	at org.gradle.internal.execution.steps.CaptureStateAfterExecutionStep.execute(CaptureStateAfterExecutionStep.java:58)
	at org.gradle.internal.execution.steps.ResolveInputChangesStep.execute(ResolveInputChangesStep.java:48)
	at org.gradle.internal.execution.steps.ResolveInputChangesStep.execute(ResolveInputChangesStep.java:36)
	at org.gradle.internal.execution.steps.BuildCacheStep.executeWithoutCache(BuildCacheStep.java:181)
	at org.gradle.internal.execution.steps.BuildCacheStep.lambda$execute$1(BuildCacheStep.java:71)
	at org.gradle.internal.Either$Right.fold(Either.java:175)
	at org.gradle.internal.execution.caching.CachingState.fold(CachingState.java:59)
	at org.gradle.internal.execution.steps.BuildCacheStep.execute(BuildCacheStep.java:69)
	at org.gradle.internal.execution.steps.BuildCacheStep.execute(BuildCacheStep.java:47)
	at org.gradle.internal.execution.steps.StoreExecutionStateStep.execute(StoreExecutionStateStep.java:36)
	at org.gradle.internal.execution.steps.StoreExecutionStateStep.execute(StoreExecutionStateStep.java:25)
	at org.gradle.internal.execution.steps.RecordOutputsStep.execute(RecordOutputsStep.java:36)
	at org.gradle.internal.execution.steps.RecordOutputsStep.execute(RecordOutputsStep.java:22)
	at org.gradle.internal.execution.steps.SkipUpToDateStep.executeBecause(SkipUpToDateStep.java:110)
	at org.gradle.internal.execution.steps.SkipUpToDateStep.lambda$execute$2(SkipUpToDateStep.java:56)
	at org.gradle.internal.execution.steps.SkipUpToDateStep.execute(SkipUpToDateStep.java:56)
	at org.gradle.internal.execution.steps.SkipUpToDateStep.execute(SkipUpToDateStep.java:38)
	at org.gradle.internal.execution.steps.ResolveChangesStep.execute(ResolveChangesStep.java:73)
	at org.gradle.internal.execution.steps.ResolveChangesStep.execute(ResolveChangesStep.java:44)
	at org.gradle.internal.execution.steps.legacy.MarkSnapshottingInputsFinishedStep.execute(MarkSnapshottingInputsFinishedStep.java:37)
	at org.gradle.internal.execution.steps.legacy.MarkSnapshottingInputsFinishedStep.execute(MarkSnapshottingInputsFinishedStep.java:27)
	at org.gradle.internal.execution.steps.ResolveCachingStateStep.execute(ResolveCachingStateStep.java:89)
	at org.gradle.internal.execution.steps.ResolveCachingStateStep.execute(ResolveCachingStateStep.java:50)
	at org.gradle.internal.execution.steps.ValidateStep.execute(ValidateStep.java:114)
	at org.gradle.internal.execution.steps.ValidateStep.execute(ValidateStep.java:57)
	at org.gradle.internal.execution.steps.CaptureStateBeforeExecutionStep.execute(CaptureStateBeforeExecutionStep.java:76)
	at org.gradle.internal.execution.steps.CaptureStateBeforeExecutionStep.execute(CaptureStateBeforeExecutionStep.java:50)
	at org.gradle.internal.execution.steps.SkipEmptyWorkStep.executeWithNoEmptySources(SkipEmptyWorkStep.java:254)
	at org.gradle.internal.execution.steps.SkipEmptyWorkStep.execute(SkipEmptyWorkStep.java:91)
	at org.gradle.internal.execution.steps.SkipEmptyWorkStep.execute(SkipEmptyWorkStep.java:56)
	at org.gradle.internal.execution.steps.RemoveUntrackedExecutionStateStep.execute(RemoveUntrackedExecutionStateStep.java:32)
	at org.gradle.internal.execution.steps.RemoveUntrackedExecutionStateStep.execute(RemoveUntrackedExecutionStateStep.java:21)
	at org.gradle.internal.execution.steps.legacy.MarkSnapshottingInputsStartedStep.execute(MarkSnapshottingInputsStartedStep.java:38)
	at org.gradle.internal.execution.steps.LoadPreviousExecutionStateStep.execute(LoadPreviousExecutionStateStep.java:43)
	at org.gradle.internal.execution.steps.LoadPreviousExecutionStateStep.execute(LoadPreviousExecutionStateStep.java:31)
	at org.gradle.internal.execution.steps.AssignWorkspaceStep.lambda$execute$0(AssignWorkspaceStep.java:40)
	at org.gradle.api.internal.tasks.execution.TaskExecution$4.withWorkspace(TaskExecution.java:281)
	at org.gradle.internal.execution.steps.AssignWorkspaceStep.execute(AssignWorkspaceStep.java:40)
	at org.gradle.internal.execution.steps.AssignWorkspaceStep.execute(AssignWorkspaceStep.java:30)
	at org.gradle.internal.execution.steps.IdentityCacheStep.execute(IdentityCacheStep.java:37)
	at org.gradle.internal.execution.steps.IdentityCacheStep.execute(IdentityCacheStep.java:27)
	at org.gradle.internal.execution.steps.IdentifyStep.execute(IdentifyStep.java:44)
	at org.gradle.internal.execution.steps.IdentifyStep.execute(IdentifyStep.java:33)
	at org.gradle.internal.execution.impl.DefaultExecutionEngine$1.execute(DefaultExecutionEngine.java:76)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeIfValid(ExecuteActionsTaskExecuter.java:139)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:128)
	at org.gradle.api.internal.tasks.execution.CleanupStaleOutputsExecuter.execute(CleanupStaleOutputsExecuter.java:77)
	at org.gradle.api.internal.tasks.execution.FinalizePropertiesTaskExecuter.execute(FinalizePropertiesTaskExecuter.java:46)
	at org.gradle.api.internal.tasks.execution.ResolveTaskExecutionModeExecuter.execute(ResolveTaskExecutionModeExecuter.java:51)
	at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:57)
	at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:56)
	at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:36)
	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.executeTask(EventFiringTaskExecuter.java:77)
	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:55)
	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:52)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:204)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:199)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:66)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:59)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:157)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:59)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.call(DefaultBuildOperationRunner.java:53)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:73)
	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter.execute(EventFiringTaskExecuter.java:52)
	at org.gradle.execution.plan.LocalTaskNodeExecutor.execute(LocalTaskNodeExecutor.java:69)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:327)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:314)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:307)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:293)
	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.execute(DefaultPlanExecutor.java:420)
	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.run(DefaultPlanExecutor.java:342)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
	at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48)
Caused by: /Users/dawidhyzy/Developer/myproject-android-tv/myproject/build/generated/assets/sentry/stagingDebug/sentry-external-modules.txt	/Users/dawidhyzy/Developer/myproject-android-tv/myproject/build/generated/assets/sentry/stagingDebug/sentry-external-modules.txt: Error: Duplicate resources
	at com.android.ide.common.resources.DataSet.checkItems(DataSet.java:512)
	at com.android.ide.common.resources.DataSet.loadFromFiles(DataSet.java:273)
	at com.android.ide.common.resources.DataSet.loadFromFiles(DataSet.java:243)
	at com.android.build.gradle.tasks.MergeSourceSetFolders.doFullTaskAction(MergeSourceSetFolders.kt:153)
	... 121 more
```

**Comment by romtsn** [comment]
Hi, could you provide a command that you started the gradle build with? Is the sentry gradle plugin applied only to the `app` module? Or are you applying it on the root project/using `subprojects` closure?

**Comment by dawidhyzy** [comment]
Sure, I used `play` button in Android Studio which I believe invokes `:myproject:assembleStagingDebug`.
It's applied only to the main app module.

**Comment by romtsn** [comment]
hmm, then I'd ask to provide a reproducible sample, if possible. I faced this issue when applying the plugin on the root project (which is incorrect), but otherwise am not sure how to reproduce it.

**Comment by dawidhyzy** [comment]
I register the plugin in the root gradle file but without applying it
```
plugins {
   [...] - other plugins
    alias(libs.plugins.sentry) apply false
}    
```
I will try to prepare something

**Comment by niltsiar** [comment]
Same problem here. It has appeared when updating AGP from `7.4.0-beta05` to the new released `7.4.0-rc01`.

I only have the plugin in the `:app` module
```
plugins {
    id "io.sentry.android.gradle" version '3.3.0'
}
```

As a workaround I've got my build running again by disabling dependencies report
```
sentry {
    includeDependenciesReport = false
}

**Comment by niltsiar** [comment]
It seems to be related to AGP yes. Now getting another duplicate resources error when trying to build a minified build
`sentry-debug-meta.properties: Error: Duplicate resources`

**Comment by pvegh** [comment]
I have the same problem when upgrading from AGP 7.4.0 beta5 to RC1.

**Comment by pvegh** [comment]
I wanted to suppress this problem somehow but today I'm getting a different error for some reason, not sure yet why it's different now.
I reported that one here: https://github.com/getsentry/sentry-android-gradle-plugin/issues/416

**Comment by markushi** [comment]
Hi @dawidhyzy, I had a look into this unfortunately couldn't reproduce the issue.
My test setup contains one app, one library.
Gradle 7.5.1
AGP 7.4.0-rc01
Sentry 6.9.1 / sentry gradle plugin 3.3.0
Both debug and release builds (including proguard) work, so I assume the issue only appears in certain scenarios / setups.

If possible please provide a reproducible sample.

**Comment by romtsn** [comment]
Maybe @pvegh or @niltsiar could help us as well and provide a reproducible sample?

**Comment by niltsiar** [comment]
For the moment I'm not able to provide a reproducible sample for this. I can say the problem it's still there for `7.4.0-rc02` and `7.4.0-rc03` but after doing some quick modifications to my project to adapt it to `8.0.0-alpha10` as said in https://github.com/getsentry/sentry-android-gradle-plugin/issues/416#issuecomment-1366195670 it seems to build. I would need to test it further to be sure but at least the build process seems not to be failing.

**Comment by markushi** [comment]
@niltsiar Thanks for getting back to us! Could you share some more details about your setup?
1. Which gradle plugins are you applying in your root/app level `build.gradle` file?
2. Are you using the Android NDK?
3. Are you using multiple gradle modules (e.g. android app module + library modules)?

**Comment by niltsiar** [comment]
Hi @markushi 

1. These are the plugins in the root `build.gradle`
```
'com.dorongold.task-tree'
```
2. These are the plugins in the app `build.gradle`
```
'com.android.application'
'com.github.ben-manes.versions'
'pmd'
'kotlin-android'
'kotlin-parcelize'
'kotlin-kapt'
'org.jlleitschuh.gradle.ktlint'
'com.google.gms.google-services'
'shot'
'io.sentry.android.gradle'
```
3. No Android NDK
4. Yes. One app module and one library module

**Comment by kahest** [comment]
For now, we need to document a workaround (disable feature) in troubleshooting pages, until we can repro this.

**Comment by romtsn** [comment]
Google issue for tracking if it makes it into 7.4 https://issuetracker.google.com/issues/261329823

**Comment by romtsn** [comment]
@dawidhyzy @niltsiar @pvegh AGP 7.4 stable is out, could you please try it out and let us know if works for you now (if the fix from 8.0.0-alpha10 made it)?

If it still doesn't work, I think we should increase the priority here and try to fix it somehow (would be great if we could still get a reproducible example for this). Thank you all.

**Comment by pvegh** [comment]
We can't try it because of the RN part. 
But it looks like the fix didn't make it to .0. Hoping for a .1 soon

**Comment by javichaques** [comment]
> @dawidhyzy @niltsiar @pvegh AGP 7.4 stable is out, could you please try it out and let us know if works for you now (if the fix from 8.0.0-alpha10 made it)?
> 
> If it still doesn't work, I think we should increase the priority here and try to fix it somehow (would be great if we could still get a reproducible example for this). Thank you all.

I confirm that the error continues with stable version 7.4.0

**Comment by chao2zhang** [comment]
This error happens with
- sentry-android-gradle-plugin 3.4
- sentry-android 6.11.0
- android gradle plugin stable version 7.4.0

**Comment by romtsn** [comment]
@chao2zhang could you also try it with AGP 8.0.0-alpha10 and let us know if it fixes the issue?

**Comment by chao2zhang** [comment]
> @chao2zhang could you also try it with AGP 8.0.0-alpha10 and let us know if it fixes the issue?

I will get it a try. It might take a few days to actually fix all other issues and test our code base against AGP 8+

**Comment by chao2zhang** [comment]
> @chao2zhang could you also try it with AGP 8.0.0-alpha10 and let us know if it fixes the issue?

I will give it a shot. It took longer than expected as I am working on a large codebase and encountered other issues when running against AGP 8+

**Comment by vovan888** [comment]
Is there any workarounds for this issue ? Disabling includeDependenciesReport does not help.

**Comment by romtsn** [comment]
Okay, I've managed to reproduce it thanks to @chao2zhang mentioning `productFlavors`. We'll look into ways fixing it or at least working around until 7.4.1 where this should be fixed on the AGP side. Sorry for the trouble everyone

**Comment by aaaamirabbas** [comment]
Hello, I have a similar problem - is there a solution for this?

**Comment by aaaamirabbas** [comment]
I tested with AGP 7.3.1, and it **worked** but not working in AGP 7.4.0

**Comment by romtsn** [comment]
just an update - I found a fix, but this requires some refactoring + using new AGP APIs, so probably will take a bit more time than just a quickfix. But hopefully should be out by the end of this week 🤞 

No known workarounds unfortunately, since the original problem is in the internals of AGP somewhere.

**Comment by digrec** [comment]
Hi and thanks for being faster than Google! Any updates? 

We have the same issue here after update to `AGP v7.4.0` last week, with:
1. `Sentry v6.12.1`
2. and `id 'io.sentry.android.gradle' version '3.4.0'` only in the `app` module.
3. and also using `productFlavors`

But we also had this warning
```
Duplicate content roots detected
Path [/Users/user/AndroidProjects/ProjectName/app/build/generated/assets/sentry/landscapeRelease] of module [Launcher.app.main] was removed from modules [Launcher.app.main]
```
even before the update, with AGP v7.3.1 and Sentry v6.7 + plugin v3.3, but no build errors then, just a warning.

AGP v7.4.0 still shows the same warning but also throws the build error:
```
> Task :app:mergeLandscapeReleaseAssets FAILED
AGPBI: {"kind":"error","text":"Duplicate resources","sources":[{"file":{"description":"sentry-debug-meta.properties","path":"/Users/user/AndroidProjects/Launcher/app/build/generated/assets/sentry/landscapeRelease/sentry-debug-meta.properties"}},{"file":{"description":"sentry-debug-meta.properties","path":"/Users/user/AndroidProjects/Launcher/app/build/generated/assets/sentry/landscapeRelease/sentry-debug-meta.properties"}}],"tool":"Resource and asset merger"}

Execution failed for task ':app:mergeLandscapeReleaseAssets'.
> /Users/user/AndroidProjects/Launcher/app/build/generated/assets/sentry/landscapeRelease/sentry-debug-meta.properties	/Users/user/AndroidProjects/Launcher/app/build/generated/assets/sentry/landscapeRelease/sentry-debug-meta.properties: Error: Duplicate resources

```

**Comment by milos-steps-app** [comment]
Hi all, @romtsn thx for the efforts. 

I can confirm same issues as @digrec, will this be addressed in the fix that is in development?

**Comment by romtsn** [comment]
the PR is up #431!

@milos-steps-app yes this should be addressed as well.

**Comment by romtsn** [comment]
apologies for holding this off so long, maven central was having issues for a week or so, so we could not publish. 

The 3.4.1 release is finally out, please try it out and let us know if there are any issues left. Thanks for your patience!

**Comment by digrec** [comment]
Thanks again for doing your best @romtsn 

Just fyi, AGP v7.3.1 + Sentry Android Plugin v3.4.1 still gives me the `Duplicate content roots detected` warning but no more build error.

Updated Gradle, AGP v7.4.0 + Sentry Android Plugin v3.4.1 and the warning is gone too! 👍

**Comment by pvegh** [comment]
I couldn't repro the original error anymore. Thanks for the fix!

**Comment by romtsn** [comment]
> Thanks again for doing your best @romtsn
> 
> Just fyi, AGP v7.3.1 + Sentry Android Plugin v3.4.1 still gives me the `Duplicate content roots detected` warning but no more build error.
> 
> Updated Gradle, AGP v7.4.0 + Sentry Android Plugin v3.4.1 and the warning is gone too! 👍

@digrec yep, the new AGP APIs are only available from 7.4.0 onwards, so the warning is expected for 7.3.1, since it still uses the old way of adding generated resources. Thanks for verifying!

**Comment by milos-steps-app** [comment]
@romtsn thx for quick reaction, much obliged
